### ISSUE TITLE
Construct/Kaitai joint PR campaign

### ISSUE THREAD

**Issue by arekbulski** [kaitai-io/kaitai_struct#328]
Would you consider, when doing the PR on 0.8 release, having Kaitai and Construct making joint posting? Construct is about to release 2.9 with a major feature (a compiler) and it would do some good (to Construct) if it would get some publicity. I am (unconditionally) going to support Kaitai whether its joint or not. 

Construct will need another 1 week before ready to ship and be ready when these are resolved:
https://github.com/construct/construct/milestone/1

**Comment by GreyCat** [comment]
"Joint PR" is probably a good thing, but we'll need to work out the details. Right now, typical KS release workflow implies:

* publishing tons of packages (probably that can't benefit Construct in any way)
* adding news entry on kaitai.io (it's possible to mention Construct there in some way, but (a) we'll need to work out how to do that, (b) the impact would be minimal — kaitai.io is not a very popular website)
* publishing link to that entry in our Twitter (it's easy to retweet your link about Construct there)
* publishing link to that entry at Reddit (Construct's announce would be separate link, so I have no idea how to help here except for upvoting your announce)
* writing some news entries for individual sites (probably they would accept 2 distinct entries for KS and Construct)

So, in other words, there's not much I can think out right out of the box: retweets, upvotes, reposting the announce, etc.

If you have any plan to write some articles or something like that — that would be potentially a much broader ground for introducing both KS and Construct to target audience, i.e. with parallel examples and stuff like that.

And, pretty please, don't call the project "Kaitai", ok? What you're referring to is "Kaitai Struct", or "KS" ;) There is at least one still unreleased project (our disassembler) that's also "Kaitai", and, potentially, several more projects, like our own YAML parsing library are likely to emerge.

**Comment by arekbulski** [comment]
I dont have any particular vision of this PR campaign. Bottom line is, I would like to present to rest of the world Kaitai and Construct as friendly competitors (not opponents). [Comrades, not arms race. ](https://www.youtube.com/watch?v=xIRPcICB-iY)

**I would be willing to write a page** (add it to Construct readthedocs) showing comparison, if thats the right word, between KS and C. Not the "mine benchmark is faster than their benchmark" kind of comparison. Just something that shows best of both frameworks. Like a tutorial, how to do basics in both. KS yaml syntax seems fucking intimidating to me, but its not unimaginable that some people using KS would see C as dunno, clunky. 

(please consider "Kaitai" a short memo for "Kaitai Struct" then :)

**Comment by KOLANICH** [comment]
BTW, why have you chosen "Kaitai"?

**Comment by arekbulski** [comment]
Well, I took over Construct about 2 years ago and recently came to a point where I concluded that the project is finished, all goals reached. So I seeked some other project where I could contribute. Someone mentioned Kaitai on my forum so I came to check it out, and got impressed too. Its clear that Kaitai and Construct have same or similar goals, parsing binary data. 

Then I came up with few ideas to take Construct to new level, a compiler and cythinization and such. But my relationship with Kaitai remains.

**Comment by KOLANICH** [comment]
@arekbulski, the question was addressed to @GreyCat , I just wondered why such a name was chosen.

**Comment by arekbulski** [comment]
Ah! Gomen, gomen. I second the question, why is it Kaitai, uncle Mika?...

**Comment by arekbulski** [comment]
Ah! Gomen, gomen. I second the question, why is it Kaitai, uncle Mika?...
"hoshii" means "want to", "kaitai" means "wishing to"
http://nihongo.anthonet.com/tai-wish-to-or-wanted-to/

**Comment by koczkatamas** [comment]
We are "wishing to Struct"? :D

**Comment by KOLANICH** [comment]
According to the link, "wishing to **buy**". But the translation doesn't lift the question "why".

**Comment by arekbulski** [comment]
I think "wishing to Struct" was much better guess. :grin:

**Comment by GreyCat** [comment]
Ok, guys, c'mon, I've repeated that thing so many times, so I thought at least @KOLANICH should have known that. I'll send you both a e-mail, ok?

**Comment by arekbulski** [comment]
Email! Email! Email!

**Comment by KOLANICH** [comment]
>I've repeated that thing so many times, so I thought at least @KOLANICH should have known that.

Sorry, I don't remember. And I think that the answer should be added to FAQ.
